CONTRIBUTING.md

```diff
@@ -1,27 +1,167 @@
+  * The PR description must explain any required documentation changes.
+  * For internal documentation in this repository, it's expected that 
+    documentation changes are delivered in the same branch.
+    Please don't abuse this tag.
+- `need-architecture-review`: is added to request a quick high-level green
+  light about a chosen approach before continuing the implementation. Also, use
+  a [draft PR][github-draft] to avoid the risk of other reviewers wasting time
+  on something that has not been agreed upon.
+- `Needs Samuele review`: Samuele (@pedronis) is our architect, and this label
+  will summon his attention. Do not use it unless you want @pedronis to review
+  your branch. If making big or deep changes, then ping Samuele in advance. The
+  tag will then be added if necessary.
+- `Needs security review`: similar to above, but with a security focus. If your
+  changes touch code in snap-confine or code related to AppArmor, Seccomp,
+  Cgroup management, then someone from the security team will be alerted and
+  will review your code.
+- `Run nested`: instructs our CI system to run our container-based
+  [nested tests][nested-tests]. These tests are usually skipped to save time,
+  but they're useful to test a PR  against certain operating system traits that
+  might otherwise be missed.
+- `Skip spread`: instructs our CI system to not run any spread tests. Only unit
+  tests will be executed. Use this only when a PR changes code in the unit tests.
+  Do not use this flag if any production code changes.
+
+### Pull request updates
+
+Feel free to [rebase][github-rebase], rework commits, and [force
+push][git-force] to your branch while a PR is waiting for its first review.
+
+However, if you are still making significant changes during this waiting
+phase, it's a good idea to keep the PR as a [draft][github-draft]. This stops
+reviewers from looking at code you may not be confident about. Set the PR as
+"Ready for review" when you do feel confident.
+
+During the review process, reviewers will point out defects or suggest
+alternative implementations.
+
+After the first review, please treat your already pushed commits as immutable
+and submit any requested changes as additional commits. This helps reviewers to
+see exactly what has changed since the last review without requesting them to
+review all the changes.
+
+Two approvals are required for a PR to be merged. A PR can then be merged into the "main" branch.
+
+On merging, you can rework the branch history as you see fit. Consider
+squashing commits from the original PR with those made during the review
+process, for example. A [force push][git-force] will be required if you do
+this.
+
+Start a [rebase][github-rebase] from the original parent commit of your first
+commit. Ensure you do not rebase on-top of the current "main" as this means
+changes from the _main_ branch will be shown in the GitHub UI as part of your
```

Comment:
Same comment as above regarding "main". Also here we have "main" quoted and then italicised. I'm not sure if that's intentional but if so I didn't get the intention